Assets/Scripts/HUD/EndGame.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Controllers;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace HUD
+{
+    public class EndGame : MonoBehaviour
+    {
+        public GameObject endGameUI;
+
+        private GameController _gameController;
+
+        public void Start()
+        {
+            _gameController = FindObjectOfType<GameController>();
```

Comment:
Not sure if this should be using Find or should be accepting a reference via the UI. It might even have a reference from the UI...